src/scripting/systemcapabilitiesbinding.cpp

```diff
@@ -0,0 +1,341 @@
+    return 1;
+}
+    
+int maxTextureUnits(lua_State* L) {
+    lua_pushnumber(L, OpenGLCap.maxTextureUnits());
+    return 1;
+}
+    
+int max2DTextureSize(lua_State* L) {
+    lua_pushnumber(L, OpenGLCap.max2DTextureSize());
+    return 1;
+}
+    
+int max3DTextureSize(lua_State* L) {
+    lua_pushnumber(L, OpenGLCap.max3DTextureSize());
+    return 1;
+}
+    
+int maxAtomicCounterBufferBindings(lua_State* L) {
+    lua_pushnumber(L, OpenGLCap.maxAtomicCounterBufferBindings());
+    return 1;
+}
+    
+int maxShaderStorageBufferBindings(lua_State* L) {
+    lua_pushnumber(L, OpenGLCap.maxShaderStorageBufferBindings());
+    return 1;
+}
+    
+int maxUniformBufferBindings(lua_State* L) {
+    lua_pushnumber(L, OpenGLCap.maxUniformBufferBindings());
+    return 1;
+}
+    
+
+} // namespace opengl
+} // namespace luascripting
+
+namespace openspace {
+namespace scripting {
+
+LuaLibrary generalSystemCapabilities() {
+    return {
+        "systemCapabilities",
+        {
+            {
+                "operatingSystem",
+                &luascripting::general::operatingSystem,
+                "",
+                "Returns a parsed string of the operating system type, for example "
+                "Windows, Linux, MacOS, or others, together with the specific version, "
+                "where available."
+            },
+            {
+                "fullOperatingSystem",
+                &luascripting::general::fullOperatingSystem,
+                "",
+                "Returns the operating system as a string. The exact format of the "
+                "returned string is implementation and operating system-dependent but it "
+                "should contain the manufacturer and the version."
+            },
+            {
+                "installedMainMemory",
+                &luascripting::general::installedMainMemory,
+                "",
+                "Returns the amount of available, installed main memory (RAM) on the "
+                "system in MB."
+            },
+            {
+                "cores",
+                &luascripting::general::cores,
+                "",
+                "Returns the number of cores."
+            },
+            {
+                "cacheLineSize",
+                &luascripting::general::cacheLineSize,
+                "",
+                "Returns the cache line size."
+            },
+            {
+                "L2Associativity",
+                &luascripting::general::L2Associativity,
+                "",
+                "Returns the L2 associativity."
+            },
+            {
+                "cacheSize",
+                &luascripting::general::cacheSize,
+                "",
+                "Returns the cache size."
+            },
+            {
+                "extensions",
+                &luascripting::general::extensions,
+                "",
+                "Returns all supported exteions as comma-separated string."
```

Comment:
Is there a reason to join the extensions to one string? How about returning a lua array-of-strings?